packages/vite/src/node/plugins/workerImportMetaUrl.ts

```diff
@@ -114,72 +105,70 @@ export function workerImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
         }
       }
       if (
-        (code.includes('new Worker') || code.includes('new ShareWorker')) &&
-        code.includes('new URL') &&
-        code.includes(`import.meta.url`)
+        !(code.includes('new Worker') || code.includes('new ShareWorker')) ||
+        !code.includes('new URL') ||
+        !code.includes(`import.meta.url`)
```

Comment:
I think this (and https://github.com/vitejs/vite/pull/7650/files#diff-30eb02919cdc99946edccdc1854b56b09fa4d784bd2ed9de30f5f9182bb6a6d9R108) aren't bad changes, but in the context of this PR it makes the change set more complex and history a little less useful. Let's revert these for now so we focus on abstracting the regex utils and the fixes.

Reply:
ok.